dotnet/Razorvine.Pyrolite/Pyrolite/Pyrolite.csproj

```diff
@@ -3,7 +3,8 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <AssemblyName>Razorvine.Pyrolite</AssemblyName>
     <RootNamespace>Razorvine</RootNamespace>
-    <LangVersion>6</LangVersion>
+    <LangVersion>7.3</LangVersion>
```

Comment:
can you say what c# 7 features have been introduced?

Reply:
> can you say what c# 7 features have been introduced?

My changes utilize the throw expressions, readonly structs, pattern matching, local functions, and out variable declaration features present in C# 7+.

Is it a concern to rely on newer language features?  If so, I can probably undo those with minimal impact on perf, the code just won't be as "clean" (by some definition of "clean" :-)).

Reply:
I had to check this: I'm using Jetbrains Rider to develop the .net code and the older version that I was using didn't support all language features, but now they do. So it's okay!

Reply:
Excellent. Thanks.